pkg/operators/machine-api-operator.go

```diff
@@ -32,100 +32,112 @@ var _ = Describe(
 		})
 
 		AfterEach(func() {
 			specReport := CurrentSpecReport()
 			if specReport.Failed() {
-				Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed())
+				Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed(), "Failed to GatherAll")
 			}
 		})
 
 		It("be available", func() {
 			ctx := framework.GetContext()
 			client, err := framework.LoadClient()
-			Expect(err).NotTo(HaveOccurred())
-			Expect(framework.IsDeploymentAvailable(ctx, client, maoDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred(), "Failed to load client")
+			Expect(framework.IsDeploymentAvailable(ctx, client, maoDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed to wait for %s Deployment to become available", maoDeployment))
 		})
 
 		It("reconcile controllers deployment", func() {
 			ctx := framework.GetContext()
 			client, err := framework.LoadClient()
-			Expect(err).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred(), "Failed to load client")
 
 			initialDeployment, err := framework.GetDeployment(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)
-			Expect(err).NotTo(HaveOccurred())
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to get %s Deployment", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q is available", maoManagedDeployment))
-			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed to wait for %s Deployment to become available", maoManagedDeployment))
 
 			By(fmt.Sprintf("deleting deployment %q", maoManagedDeployment))
-			Expect(framework.DeleteDeployment(ctx, client, initialDeployment)).NotTo(HaveOccurred())
+			Expect(framework.DeleteDeployment(ctx, client, initialDeployment)).NotTo(HaveOccurred(),
+				fmt.Sprintf("Failed to delete %s Deployment", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q is available again", maoManagedDeployment))
-			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentAvailable(ctx, client, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed to wait for %s Deployment to become available", maoManagedDeployment))
 
 			By(fmt.Sprintf("checking deployment %q spec matches", maoManagedDeployment))
-			Expect(framework.IsDeploymentSynced(ctx, client, initialDeployment, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue())
+			Expect(framework.IsDeploymentSynced(ctx, client, initialDeployment, maoManagedDeployment, framework.MachineAPINamespace)).To(BeTrue(),
+				fmt.Sprintf("Failed verifying %s Deployment spec has been reconciled", maoDeployment))
```

Comment:
should this be `maoManagedDeployment` ?

```suggestion
				fmt.Sprintf("Failed verifying %s Deployment spec has been reconciled", maoManagedDeployment))
```